python/python/bystro/utils/tar.py

```diff
@@ -0,0 +1,76 @@
+"""Utility functions for safely invoking tar in cross-OS manner."""
+
+import subprocess
+from enum import Enum
+from subprocess import CalledProcessError
+from typing import NoReturn
+
+
+class OperatingSystem(Enum):
+    """Represent the operating system on which the current Python process is running."""
+
+    linux = "linux"
+    macosx = "macosx"
+
+
+# here and throughout, fully qualify executable names to avoid privilege escalation
+
+
+def _get_executable_name(name: str) -> str:
+    try:
+        result = subprocess.run(
+            ["/usr/bin/which", name],
+            capture_output=True,
+            text=True,
+            check=True,
+            shell=False,  # noqa: S603 (input is safe)
+        )
+    except CalledProcessError as err:
+        err_msg = f"executable `{name}` not found on system, is it installed?"
+        raise OSError(err_msg) from err
+    return result.stdout.strip()
+
+
+def assert_never(value: NoReturn) -> NoReturn:
+    """Raise error for mypy if code is proven reachable."""
+    msg = f"Unhandled value: {value} ({type(value).__name__})"
+    raise AssertionError(msg)
+
+
+def _determine_os() -> OperatingSystem:
+    """Determine whether we're running on Linux or MacOSX by inspecting uname output."""
+    try:
+        result = subprocess.run(
+            ["/usr/bin/uname", "-a"],  # noqa: S603 (input is safe)
```

Comment:
Could we use sys.platform instead? https://docs.python.org/3/library/sys.html#sys.platform

More portable